src/sentry/static/sentry/app/components/organizations/globalSelectionHeader/index.jsx

```diff
@@ -199,14 +207,39 @@ class GlobalSelectionHeader extends React.Component {
       return true;
     }
 
+    // Update if `forceUrlSync` changes
+    if (this.props.forceUrlSync !== nextProps.forceUrlSync && !this.props.forceUrlSync) {
+      return true;
+    }
+
     return false;
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.hasCustomRouting) {
+    const {hasCustomRouting, location, forceUrlSync, selection} = this.props;
+
+    if (hasCustomRouting) {
       return;
     }
 
+    // Kind of gross
+    if (forceUrlSync !== prevProps.forceUrlSync && !prevProps.forceUrlSync) {
```

Comment:
`!forceUrlSync && prevProps.forceUrlSync` is a little more readable, similar in componentDidUpdate